src/shims/unix/macos/foreign_items.rs

```diff
@@ -174,6 +174,70 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(res, dest)?;
             }
 
+            "os_unfair_lock_lock" => {
+                let [lock_op] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let id =
+                    this.mutex_get_or_create_id(lock_op, this.libc_ty_layout("os_unfair_lock"), 0)?;
+
+                if this.mutex_is_locked(id) {
+                    if this.mutex_get_owner(id) == this.active_thread() {
+                        throw_machine_stop!(TerminationInfo::Abort(
+                            "attempted to lock an os_unfair_lock that is already locked by the current thread".to_owned()
+                        ));
+                    }
+
+                    this.mutex_enqueue_and_block(id, None, dest.clone());
+                } else {
+                    this.mutex_lock(id);
+                }
+            }
+            "os_unfair_lock_trylock" => {
+                let [lock_op] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let id =
+                    this.mutex_get_or_create_id(lock_op, this.libc_ty_layout("os_unfair_lock"), 0)?;
+
+                if this.mutex_is_locked(id) {
+                    this.write_scalar(Scalar::from_bool(false), dest)?;
```

Comment:
So this does *not* detect reentrancy?

Reply:
No, it doesn't, it simply returns `false` in all cases where the lock couldn't be acquired.